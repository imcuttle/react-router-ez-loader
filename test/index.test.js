/**
 * @file: index.test.js
 * @author: Cuttle Cong
 * @date: 2018/1/11
 * @description:
 */
const loader = require('../index')
const fs = require('fs')
const source = fs.readFileSync(require.resolve('./fixture/routes')).toString()

test('Test', () => {
  expect(loader(source)).toMatchSnapshot()
})
