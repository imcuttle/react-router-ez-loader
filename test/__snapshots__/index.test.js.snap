// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1`] = `
"/**
 * @file: routes
 * @author: Cuttle Cong
 * @date: 2018/1/11
 * @description:
 */

module.exports = {
  path: '/',
  component: require(\\"./App\\").default || require(\\"./App\\"),
  indexRoute: {
    getComponent: function getComponent(location, callback) {
      require.ensure([], function () {
        callback(null, require(\\"@IndexPage\\").default || require(\\"@IndexPage\\"));
      }, \\"@IndexPage\\");
    }
  },
  childRoutes: [{
    path: 'edit',
    getComponent: function getComponent(location, callback) {
      require.ensure([], function () {
        callback(null, require(\\"!some-loader!../../Edit\\\\\\"Page\\").default || require(\\"!some-loader!../../Edit\\\\\\"Page\\"));
      }, \\"Edit\\\\\\"Page\\");
    }
  }, {
    path: 'view',
    getComponent: function getComponent(location, callback) {
      require.ensure([], function () {
        callback(null, require(\\"!some-loader!../../ViewPage?n=View\\").default || require(\\"!some-loader!../../ViewPage?n=View\\"));
      }, \\"View\\");
    }
  }]
};"
`;
